mkContainer/mkContainer.nix

```diff
@@ -8,72 +8,172 @@
+    optionalAttrs (isNixStoreUserOwned) {
+      inherit (nixStoreUserGroup)
+        uname
+        gname
+        uid
+        gid
+        ;
+
+      # chown the /run directory to the nixStoreOwner, so that Nix can run as a
+      # single user installation inside the container
+      fakeRootCommands = ''
+        chown -R ${toString nixStoreUserGroup.uid}:${toString nixStoreUserGroup.gid} /run
+      '';
+      enableFakechroot = true;
+    }
+    // {
+      name = containerName;
+      tag = containerTag;
+      created = containerCreated;
+
+      # Ensures the container configuration contains the correct architecture of
+      # the binaries contained within it. Runtimes can use this to short-circuit
+      # errors when users try to run a container on an incompatible architecture.
+      architecture = containerPkgs.go.GOARCH;
+
+      # Activate script requires writable directory, /run feels like a logical place.
+      extraCommands = ''
+        mkdir -m 1770 run
+        mkdir -p -m 1770 run/flox
+      '';
+
+      # symlinkJoin fails when drv contains a symlinked bin directory, so wrap in an additional buildEnv.
+      contents = pkgs.buildEnv {
+        name = "contents";
+        paths = [
+          fakeNss
+          environment
+          (lowPrio containerPkgs.bashInteractive) # for a usable shell
+          (lowPrio containerPkgs.coreutils) # for just the basic utils
+        ];
+      };
+      config = containerConfig // {
+        # Use activate script as the [one] entrypoint capable of
+        # detecting interactive vs. command activation modes.
+        # Usage:
+        #   podman run -it
+        #     -> launches interactive shell with controlling terminal
+        #   podman run -i <cmd>
+        #     -> invokes interactive command
+        #   podman run -i [SIC]
+        #     -> launches crippled interactive shell with no controlling
+        #        terminal .. kinda useless
+        Entrypoint = [ "${environment}/activate" ];
+
+        Env = mapAttrsToList (name: value: "${name}=${value}") {
+          "FLOX_ENV" = environment;
+          "FLOX_PROMPT_ENVIRONMENTS" = "floxenv";
+          "FLOX_PROMPT_COLOR_1" = "99";
+          "FLOX_PROMPT_COLOR_2" = "141";
+          "_FLOX_ACTIVE_ENVIRONMENTS" = "[]";
+          "FLOX_SOURCED_FROM_SHELL_RC" = "1"; # don't source from shell rc (again)
+          "_FLOX_FORCE_INTERACTIVE" = "1"; # Required when running podman without "-t"
+          "FLOX_SHELL" = "${containerPkgs.bashInteractive}/bin/bash";
+          "FLOX_RUNTIME_DIR" = "/run/flox";
+        };
+      };
+
+      passthru = {
+        # This tests can be ran with the following command from the root of the repository:
+        #     $ nix eval --impure --expr '(import ./mkContainer/mkContainer.nix { nixpkgsFlakeRef = "github:nixos/nixpkgs?ref=nixos-24.11"; environmentOutPath = null; system = builtins.currentSystem; containerSystem = builtins.currentSystem; }).passthru.tests'
```

Comment:
Nice! I think that's good for now - it documents all the cases we need to keep in mind if we ever change this